kale/loaddata/polypharmacy_datasets.py

```diff
@@ -0,0 +1,54 @@
+import os
+
+import torch
+from torch.utils.data import Dataset
+from torch_geometric.data.data import Data
+from yacs.config import CfgNode
+
+from kale.utils.download import download_file_by_url
+
+
+class PolypharmacyDataset(Dataset):
+    r"""Polypharmacy side effect prediction dataset. Only for full-batch training.
+
+    Args:
+        cfg_dataset (CfgNode): configurations of the dataset.
+        mode (str): "train" or "test". Defaults to "train".
+    """
+
+    def __init__(self, cfg_dataset: CfgNode, mode: str = "train"):
+
+        super(PolypharmacyDataset, self).__init__()
+
+        self.cfg_dataset = cfg_dataset
+        data = self.load_data()
+
+        self.edge_index = data.__getitem__(f"{mode}_idx")
+        self.edge_type = data.__getitem__(f"{mode}_et")
+        self.edge_type_range = data.__getitem__(f"{mode}_range")
+
+        if mode == "train":
+            self.protein_feat = data.g_feat
+            self.protein_edge_index = data.gg_edge_index
+            self.drug_feat = data.d_feat
+            self.protein_drug_edge_index = data.gd_edge_index
```

Comment:
I have thought we correct the above variable named. `g_`, `gg_`, `d_`, `gd_`. Is it else where? Can we make them more descriptive here? Also gene or protein, again. I saw you used protein more.

Reply:
This is due to the variable name in the dataset, to be updated in another PR.